[Impeller] add doc on iOS flamegraph capture. (#49469)
flutter/engine: impeller/docs/ios_cpu_profile.md

# iOS CPU Profiling

XCode has a number of built in profiling tools that can easily be used with local or prebuilt engines. This document is focused on creating flame graph charts, which is not a part of XCode but provides a useful way to look at aggregate performance. Compared to the Android Studio flame graph this one is a little wonky, but still useful.

1. Clone the "FlameGraph" swift package from `https://github.com/lennet/FlameGraph`
2. Start a "Time Profiler" based profile session.

![Alt text](assets/ios_profiling/ios_time_profiler_example.png)

  Click the red dot to begin and record as much profile data as you want, then press stop to conclude the trace.

3. Select the thread to investigate, in this case you want io.flutter.raster. IMPORTANT: also select the trace root.

![Alt text](assets/ios_profiling/ios_profiler_select_thread.png)

> 4. Either via keyboard short cut or edit menu, "Deep Copy" the trace.

Reviewer: ```suggestion
4. Copy the trace with the keyboard shortcut or the menu ("Edit" -> "Deep Copy").
```

Reviewer: If we mention the keyboard shortcut we should mention what it is.

Author: Done
